File: 1346. Check If N and Its Double Exist.cpp
class Solution {
public:
    bool checkIfExist(vector<int>& arr) 
    {
        int n = arr.size();
        for(int i=0;i<n;i++)
        {
            for(int j=0;j<n;j++)
                if(i != j and arr[j]*2 == arr[i]) 
                    return true;
        }
        return false;
    }
};



class Solution {
public:
    bool checkIfExist(vector<int>& arr) 
    {
        map<int,int>mp;
        mp[0] = -1;
        for(auto val:arr) mp[val]++;
        for(auto val:arr)
        {
            if(val%2) 
            {
                if(mp[val+val]) return true;
            }
            else if(mp[val/2]) return true;
        }
        return false;
    }
};

File: 1346. Check If N and Its Double Exist.py
class Solution:
    def check(self, arr):
        store = defaultdict(int)
        for val in arr:
            print(val * 2)
            if store[val * 2]:
                return True
            store[val] = 1
        return False

    def checkIfExist(self, arr: List[int]) -> bool:
        arr.sort()
        if self.check(arr):
            return True
            
        arr.reverse()
        return self.check(arr)